tests/testhelpers.py

```diff
@@ -1,257 +1,200 @@
+    baselines.Fisher, baselines.Herbalist,
+    baselines.Prospector,baselines.Carver, baselines.Alchemist,
+    baselines.Melee, baselines.Range, baselines.Mage]
+
+
+# pylint: disable=abstract-method,duplicate-code
+class ScriptedAgentTestEnv(nmmo.Env):
+  '''
+  EnvTest step() bypasses some differential treatments for scripted agents
+  To do so, actions of scripted must be serialized using the serialize_actions function above
+  '''
+  __test__ = False
+
+  def __init__(self, config: nmmo.config.Config, seed=None):
+    super().__init__(config=config, seed=seed)
+    # this is to cache the actions generated by scripted policies
+    self.actions = {}
+
+  def reset(self, map_id=None, seed=None, options=None):
+    self.actions = {}
+    # manually resetting the EntityState, ItemState datastore tables
+    EntityState.State.table(self.realm.datastore).reset()
+    ItemState.State.table(self.realm.datastore).reset()
+    return super().reset(map_id=map_id, seed=seed, options=options)
+
+  def _compute_scripted_agent_actions(self, actions):
+    # all agent must be scripted agents when using ScriptedAgentTestEnv
+    for ent in self.realm.players.values():
+      assert isinstance(ent.agent, baselines.Scripted), 'All agent must be scripted.'
+
+    # if actions are not provided, generate actions using the scripted policy
+    if actions == {}:
+      self.actions = {}
+      for eid, ent in self.realm.players.items():
+        # generate the serialized actions & cache these
+        atns = ent.agent(self.obs[eid])
+
+        # handle the cases that are problematic for pickle
+        for atn, args in atns.items():
+          for arg, val in args.items():
+            if arg == nmmo.io.action.Price and not isinstance(val, int):
+              # <class 'nmmo.io.action.Price'>: <class 'nmmo.io.action.Discrete_1'>
+              # convert Discrete_1 to 1
+              atns[atn][arg] = val.val
+
+        # make a copy of actions that are not serialized
+        self.actions[eid] = deepcopy(atns)
+
+        #print(eid, self.actions[eid])
+
+        # deserialize
+        for atn, args in atns.items():
+          for arg, val in args.items():
+            #print(ent, val)
+            atns[atn][arg] = arg.deserialize(self.realm, ent, val)
+        actions[eid] = atns
+
+    # if actions are provided, deserialize these
+    else:
+
+      # WARNING: this is a hack to set the random number generator to the same state
+      # since scripted agents also use RNG. Without this, the RNG is in different state,
+      # and the env.step() does not give the same results.
+      for eid, ent in self.realm.players.items():
+        ent.agent(self.obs[eid])
+
+      # Now, process the received actions
+      self.actions = deepcopy(actions)
+      actions = {}
+      for eid in self.actions.keys():
+        assert eid in self.realm.players, f'Entity {eid} not in realm'
+        ent = self.realm.players[eid]
+        atns = deepcopy(self.actions[eid])
```

Comment:
can we move this into _deserialize_actions()

Reply:
yap, moved into env._deserialize_scripted_actions()